### PR TITLE
upgrade vault api and support kv2

### PR DESCRIPTION
**Pull request by daqingshu**

### CODE REVIEW

File: vault.go [hunk 1]
```diff
@@ -25,7 +25,11 @@ func getCredentialSetFromVault(host string) (map[string]string, error) {
 	}
 
 	var out = map[string]string{}
-	for k, v := range secret.Data {
+	data, ok := secret.Data["data"].(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf("secret Data does not contain data")
+	}
+	for k, v := range data {
```
Comment (Luzifer) [inline vault.go]: This breaks K/V v1 compatibility.

**Review by Luzifer — CHANGES_REQUESTED** [review]

File: vault.go (hunk 1 above)
Comment (daqingshu) [reply vault.go]: It is indeed incompatible, the vault doc https://www.vaultproject.io/docs/secrets/kv/kv-v2#acl-rules 
should a git-credential-vault v2 be release?

Comment (Luzifer) [reply vault.go]: As long as both K/V versions are supported by Vault they should be supported here too…

So basically if I see it correctly it should be this:
```
if secret.Data.data is map[string]interface{}:
	use secret.Data.data
elif secret.Data is map[string]interface{}:
	use secret.Data
else:
	error
```

At least I don't see any other difference which should be relevant for us… 🤔

Comment (daqingshu) [reply vault.go]: updated, please review, tks a lot.

File: vault.go [hunk 2]
```diff
@@ -24,8 +24,14 @@ func getCredentialSetFromVault(host string) (map[string]string, error) {
...
 
+	var data map[string]interface{}
 	var out = map[string]string{}
-	for k, v := range secret.Data {
+	var ok bool
+	data, ok = secret.Data["data"].(map[string]interface{})
+	if !ok {
+		data = secret.Data
+	}
+	for k, v := range data {
```
Comment (Luzifer) [inline vault.go]: Please simplify the three variable declarations in a single `var ( ... )` block and use shorter `if data, ok = ...; !ok {` syntax for condition

Comment (daqingshu) [reply vault.go]: ok, updated, plz review tks

**Review by Luzifer — APPROVED** [review]

**Comment by Luzifer** [comment]
Thank you very much for the contribution!

**Comment by daqingshu** [comment]
> Thank you very much for the contribution!

It's my pleasure :)

**Pull request merged**